flink-table/flink-table-planner-blink/src/main/scala/org/apache/flink/table/planner/codegen/GenerateUtils.scala

```diff
@@ -370,8 +369,17 @@ object GenerateUtils {
         generateNonNullLiteral(literalType, literalValue.toString, literalValue)
 
       case TIMESTAMP_WITHOUT_TIME_ZONE =>
-        val millis = literalValue.asInstanceOf[Long]
-        generateNonNullLiteral(literalType, millis + "L", millis)
+        val fieldTerm = newName("timestamp")
+        val timestampString = literalValue.asInstanceOf[TimestampString].toString
+        val millis = getMillisSinceEpoch(timestampString)
```

Comment:
You should just use `TimestampString` to `localDateTime`.

Reply:
How can `TimestampString` get a `LocalDateTime`?

Reply:
In `getMillisSinceEpoch`, you have got the year, month, and etc.. Why not construct a `LocalDateTime`?

Reply:
`getMillisSinceEpoch` is a util function of `TimestampString`, we copied it because our copied `DateTimeUtils` causes Gregorian cutovers issues. We should use `TimestampString.getMillisSinceEpoch` directly when we remove our copied `DateTimeUtils`. Do you mean we should not depend on `TimestampString` and parse the timestamp string by ourself?